ProjFS.Mac/PrjFSKext/KauthHandler.cpp

```diff
@@ -129,7 +143,19 @@ kern_return_t KauthHandler_Init()
     {
         goto CleanupAndFail;
     }
-    
+
+
+    s_renameLock = SpinLock_Alloc();
+    s_maxPendingRenames = 8; // Arbitrary choice, should be maximum number of expected concurrent threads performing renames, but array will resize on demand
```

Comment:
I could see some applications doing a bunch of renames (this is how a lot of editors seem to save).   Do we know how many threads the Kernel runs with?

Reply:
There is a kernel thread for every thread in every process, plus it can create its own. So there are usually thousands. Very few will be between the kauth fileop and vnode authorisation calls in the rename code at any time though, and if there are more than 8, well, that's why allow resizing.